api/src/main/java/org/apache/cloudstack/api/command/admin/vm/MigrateVirtualMachineWithVolumeCmd.java

```diff
@@ -152,20 +153,17 @@ public ApiCommandResourceType getApiResourceType() {
     @Override
     public void execute() {
         if (hostId == null && MapUtils.isEmpty(migrateVolumeTo)) {
-            throw new InvalidParameterValueException(String.format("Either %s or %s  must be passed for migrating the VM", ApiConstants.HOST_ID, ApiConstants.MIGRATE_TO));
-        }
-
-        UserVm userVm = _userVmService.getUserVm(getVirtualMachineId());
-        if (userVm == null) {
-            throw new InvalidParameterValueException("Unable to find the VM by id=" + getVirtualMachineId());
+            throw new InvalidParameterValueException(String.format("Either %s or %s  must be passed for migrating the VM.", ApiConstants.HOST_ID, ApiConstants.MIGRATE_TO));
         }
 
+        VirtualMachine userVm = _userVmService.getVm(getVirtualMachineId());
```

Comment:
how about the `null` check that used to be there? in `UserVmManagerImpl.getVM(long)` I see no sign of `null`-safety.

Reply:
As I was adjusting the error messages, I could not verify this one. Furthermore, I verified that the method `ParamProcessWorker#translateUuidToInternalId` already makes this verification. If the user puts an nonexistent UUID, or and UUID from another entity, the `translateUuidToInternalId` method already throws an exception, for these reason I removed this `null` check.